src/foam/nanos/auth/PermissionTableView.js

```diff
@@ -94,7 +97,17 @@ foam.CLASS({
                               cb.data$.sub(function() { self.updateGroup(p, g, cb.data); });
                               this.start('td').show(groups[g.id].data$.map(function() {return groups[g.id].data;}))
                                     .style({'text-align': 'center', 'width': '100'}).tag(cb).call(function() {
-                                      if ( g.implies(p.id) ) { cb.style({'border-color': '#40C75B'}) };
+                                      self.groupDAO.find(g.parent).then(function(a){
+                                        if ( a != undefined && a.implies(p.id) ) {
+                                          cb.setAttribute('title', g.parent + ': ' + p.id); //'Permission inherited from ' + g.parent
+                                          cb.style({'border-color': '#40C75B'})
+                                        }
+                                      });
+
+                                      if ( g.implies(p.id) ) {
+                                        cb.setAttribute('title', g.id + ': ' + p.id);  //'Permissions granted from ' + g.id
+                                        cb.style({'border-color': '#40C75B'});
+                                      }
```

Comment:
Move the parent group code above into the else block of this 'if' to avoid extra work and avoid a race condition.